### ISSUE TITLE
[Feature] Log metrics in test mode

### ISSUE THREAD

**Issue by mmeendez8** [open-mmlab/mmengine#1482]
### What is the feature?

Just noticed that when running mmseg with a pretrained model and a test set to evaluate its performance, the final metrics are not logged in my visbackend (mlflow). 

I was exploring the source and noticed that the `LoggerHook` class is the one in charge of dumping metrics during [training ](https://github.com/open-mmlab/mmengine/blob/f79111ecc0eea9fbb1b7d1361a79f7062ca1ac10/mmengine/hooks/logger_hook.py#L199) and [eval](https://github.com/open-mmlab/mmengine/blob/f79111ecc0eea9fbb1b7d1361a79f7062ca1ac10/mmengine/hooks/logger_hook.py#L273).

I was wondering if there is any reason of why `runner.visualizer.add_scalars()` is not called [`after_test_epoch`](https://github.com/open-mmlab/mmengine/blob/f79111ecc0eea9fbb1b7d1361a79f7062ca1ac10/mmengine/hooks/logger_hook.py#L276)

**Comment by mmeendez8** [comment]
@HAOCHENYE would it be possible to get some feedback on this?

**Comment by fsbarros98** [comment]
Also need update on this!

**Comment by HAOCHENYE** [comment]
Sorry for the late response. The reason for not calling add_scalars in [after_test_epoch](https://github.com/open-mmlab/mmengine/blob/f79111ecc0eea9fbb1b7d1361a79f7062ca1ac10/mmengine/hooks/logger_hook.py#L276) is that the test set typically does not have the ground truth, and we usually only calculate various metrics and statistics on the validation set.

**Comment by fsbarros98** [comment]
that is a somewhat valid response, but especially if I'm running a test.py script I would expect that test metrics would be logged in

**Comment by mmendezz** [comment]
I see... so is your plan to assume test set does not have the ground truth or should we find a way of extracting and logging them when gt is present?

**Comment by mmeendez8** [comment]
I see... so is your plan to assume test set does not have the ground truth or should we find a way of extracting and logging them when gt is present?

**Comment by fsbarros98** [comment]
If gt is not present do we even have metrics? I'm using this for MMagic, so in generation we also don't have gt, but we mainly compute metrics by comparing test features (generated samples) and train features... I believe that whenever metrics are calculated for testing they should also be added to the visualizer

**Comment by HAOCHENYE** [comment]
Visualizer is a globally accessible variable, and you can get the visualizer at any location using `visualizer = Visualizer.get_current_instance()` and then call the interface like `visualizer.add_scalar()` to record the information you want. You can implement it in a custom hook, or any other places you want (maybe `model.xxx`, `metric.xxx` ...)
